text/0118-search-api.md

```diff
@@ -0,0 +1,346 @@
+
+- 🔴 Sending a value with a different type than `Array of String`(POST), `String`(GET) or `null` for `attributesToRetrieve` will returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 1.2.1.8 `attributesToHighlight`
+
+- Type: Array[String](POST)|String(GET)
+- Required: False
+- Default: `[]|null`
+
+Highlights matching query terms in the specified attributes by enclosing them in `<em>` tags.
+
+When this parameter is set, returned documents include a `_formatted` object containing the highlighted terms.
+
+If `"*"` is provided as a value: `attributesToHighlight=["*"]` all the attributes present in `attributesToRetrieve` will be assigned to `attributesToHighlight`.
+
+- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToHighlight` will returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+> See [_Formatted Field Behavior](0039-_formatted-field-behavior_.md)
+
+##### 1.2.1.9 `attributesToCrop`
+
+- Type: Array[String]|String
+- Required: False
+- Default: `[]|null`
+
+Crops the selected attributes' values in the returned results to the length indicated by the `cropLength` parameter.
+
+When this parameter is set, returned documents include a `_formatted` object containing the cropped terms.
+
+Optionally, indicating a custom crop length for any of the listed attributes is possible: `attributesToCrop=["attributeNameA:25", "attributeNameB:150"]`. The custom crop length set in this way has priority over the `cropLength` parameter.
+
+Instead of supplying individual attributes, it is possible to provide `["*"]` as a value: `attributesToCrop=["*"]`. This will crop the values of all attributes present in `attributesToRetrieve`.
+
+- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToCrop` will returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+> See [_Formatted Field Behavior](0039-_formatted-field-behavior_.md)
+
+##### 1.2.1.10 `cropLength`
+
+- Type: Integer
+- Required: False
+- Default: `200`
+
+Configures the number of characters to keep on each side of the matching query term when using the `attributesToCrop` parameter.
+
+If `attributesToCrop` is not configured, `cropLength` has no effect on the returned results.
+
+- 🔴 Sending a value with a different type than `Integer` or `null` for `cropLength` will returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 1.2.1.11 `matches`
+
+- Type: Boolean
+- Required: False
+- Default: `false`
+
+Adds a `_matchesInfo` object to the search response that contains the location of each occurrence of queried terms across all fields. This is useful when more control is needed than offered by the built-in highlighting/cropping features.
```

Comment:
Maybe give some example of "when more control is needed than offered by the built-in highlighting/cropping features"

Reply:
See https://github.com/meilisearch/specifications/pull/118/files#r816114252